packages/recurring-payments-job/src/recurring-payments-processor.js

```diff
@@ -16,17 +17,30 @@ export const processRecurringPayments = async () => {
 
 const processRecurringPayment = async record => {
   const referenceNumber = record.expanded.activePermission.entity.referenceNumber
+  const agreementId = record.entity.agreementId
+  const transaction = await createNewTransaction(referenceNumber)
+  await takeRecurringPayment(agreementId, transaction)
+}
+
+const createNewTransaction = async referenceNumber => {
   const transactionData = await processPermissionData(referenceNumber)
   console.log('Creating new transaction based on', referenceNumber)
   try {
     const response = await salesApi.createTransaction(transactionData)
     console.log('New transaction created:', response)
+    return response
   } catch (e) {
     console.log('Error creating transaction', JSON.stringify(transactionData))
     throw e
   }
 }
 
+const takeRecurringPayment = async (agreementId, transaction) => {
+  const preparedPayment = preparePayment(agreementId, transaction)
+  console.log('Requesting payment:', preparedPayment)
+  await sendPayment(preparedPayment)
```

Comment:
I'm not sure why you're awaiting this - we're not interested in the response, so there's no need to await it. We can fire and forget.

Reply:
Oops, this is a holdover from an earlier async version – fixed!